File: 0007-reverse-integer/0007-reverse-integer.py
class Solution:
    def reverse(self, x: int) -> int:
        s=str(x)
        if s[0] == '-':
         num= int('-' + s[-1:0:-1])
         if num >= -2147483648 and num<= 2147483647:
            return num
         else:
            return 0
        else:
            num = int(s[::-1])
            if num >= -2147483648 and num<= 2147483647:
               return num
            else:
               return 0

File: 0007-reverse-integer/0007-reverse-integer.java
class Solution {
    public int reverse(int x) {
        int val=0;
        int revnum = 0;
        if (x < 0) {
            x *= -1;
            val= reversing(x,revnum)*-1;
        } else {
            val=reversing(x,revnum);

        }
        if(val<Math.pow(-2,31) || val>Math.pow(2,31)-1){
            return 0;
        }
        else return val;
    }
    public int reversing(int x,int revnum){
        while (x > 0) {
            if(revnum>Integer.MAX_VALUE/10 || revnum<Integer.MIN_VALUE/10){
                return 0;
            }
            int last = x % 10;
            x /= 10;
            revnum = (revnum * 10) + last;
        }
        return revnum;
    }
}